pkg/apis/core/v1alpha1/hardware_types.go

```diff
@@ -130,6 +129,7 @@ type DHCP struct {
 	UEFI        bool     `json:"uefi,omitempty"`
 	IfaceName   string   `json:"iface_name,omitempty"`
 	IP          *IP      `json:"ip,omitempty"`
+	VLANID      uint     `json:"vlan_id,omitempty"`
```

Comment:
Any particular reason we're using unsigned?

Reply:
valid VLANs are from 0-4096 so i wanted to eliminate negative numbers. This makes me think i should add a regex check to this affect.

Reply:
Thinking about this, i actually want this to be a string. so that the default value is not 0 but “”.

Reply:
I've changed it to be a string and added a regex validation.